### PR TITLE
reaper: darwin platform

### PR DESCRIPTION
**Pull request by viraptor**
## Description of changes

Add darwin package installation.
I'm ignoring the possibility that this is going to be run on a pre-10.15 system which requires a different dmg download.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/reaper/default.nix [hunk 1]
```diff
@@ -32,31 +35,50 @@ stdenv.mkDerivation rec {
...
+  ] ++ lib.optionals stdenv.isDarwin [
+    undmg
   ];
 
+  # default unpack complains about multiple directories
+  unpackPhase = lib.optionalString stdenv.isDarwin ''
+    runHook preUnpack
+    undmg "$src"
+    runHook postUnpack
+  '';
```
Comment (wegank) [inline pkgs/applications/audio/reaper/default.nix:57]: What about setting `sourceRoot`?

File: pkgs/applications/audio/reaper/default.nix [hunk 2]
```diff
@@ -14,14 +15,16 @@
...
+, jackSupport ? stdenv.isLinux
 , jackLibrary
-, pulseaudioSupport ? config.pulseaudio or true
+, pulseaudioSupport ? config.pulseaudio or stdenv.isLinux
 , libpulseaudio
 }:
 
 let
-  url_for_platform = version: arch: "https://www.reaper.fm/files/${lib.versions.major version}.x/reaper${builtins.replaceStrings ["."] [""] version}_linux_${arch}.tar.xz";
+  url_for_platform = version: arch: if stdenv.isDarwin
```
Comment (wegank) [inline pkgs/applications/audio/reaper/default.nix:25]: Since it's universal, I assume it has `x86_64-darwin` support?

Comment (viraptor) [reply pkgs/applications/audio/reaper/default.nix:25]: Possibly. I don't have hardware to check, but I could yolo add it and hope for the best.

File: pkgs/applications/audio/reaper/default.nix (hunk 1 above)
Comment (viraptor) [reply pkgs/applications/audio/reaper/default.nix]: Yeah, that works.

File: pkgs/applications/audio/reaper/default.nix (hunk 2 above)
Comment (viraptor) [reply pkgs/applications/audio/reaper/default.nix:25]: It does work, but I struggle with updating the hashes in the file. `update-source-version` really does not like it if two archs have the same hash :(

Comment (viraptor) [reply pkgs/applications/audio/reaper/default.nix:25]: ok, the last update works

**Comment by wegank** [comment]
Result of `nixpkgs-review pr 266834` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>reaper</li>
  </ul>
</details>

**Pull request merged**